prow/tide/tide.go

```diff
@@ -1146,6 +1147,16 @@ func (c *Controller) mergePRs(sp subpool, prs []PullRequest) error {
 			continue
 		}
 
+		// Ensure tide context has success state, otherwise PR merge will fail if branch protection
+		// in github is enabled and the loop to change tide context hasn't done it already
+		err = setTideStatusSuccess(pr, c.ghc, c.config(), log)
```

Comment:
Without synchronization with the status controller there will be a race here. The status controller could set the status back to pending before or after we try to merge.

Reply:
Hi thanks for reviewing, were you thinking maybe set another gh context 'tide-merging' and have the loop ignore those PRs? Or how is this 'state' regularly handled in prow

Reply:
Adding another GH context would not be appropriate for this. We only need once context, we just need the status and sync controllers to be in agreement.
The state of the status context is currently entirely managed by the tide status controller which is a separate set of goroutines from the tide sync controller that handles retesting and merging. If the sync controller needs a status to be set to green immediately before merge, it needs to synchronize with the status controller to either 1) have the status controller set the context on its behalf or 2) ensure the status controller is not and will not interfere with the status on that PR so that the sync controller can set the status to green itself.